src/lib/hooks/swap/useSwapCallback.tsx

```diff
@@ -17,18 +18,33 @@ export enum SwapCallbackState {
   VALID,
 }
 
+interface UseSwapCallbackReturns {
+  state: SwapCallbackState
+  callback: null | (() => Promise<TransactionResponse>)
+  error: ReactNode | null
+}
+interface UseSwapCallbackArgs {
+  trade: AnyTrade | undefined // trade to execute, required
+  allowedSlippage: Percent // in bips
+  recipientAddressOrName: string | null | undefined // the ENS name or address of the recipient of the trade, or null if swap should be returned to sender
+  signatureData: SignatureData | null | undefined
+  deadline: BigNumber | undefined
+  fee?: FeeOptions
```

Comment:
Use consistent typing. fee should be `FeeOptions | undefined`, or else the other optional inputs should be marked as optional.

Reply:
I would prefer they all be `X | undefined` so that they are passed, even if undefined, and future updates don't miss them because they are optional.

Reply:
> future updates don't miss them because they are optional

I'm not following here